### PR TITLE
Gizmo-challenge completed

### PR DESCRIPTION
**Pull request by tpfau**
I hope I didn't miss anything.

### CODE REVIEW

File: gizmo.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+Created on Mar 31, 2022
+Gizmo Challenge File
+
+@author: thomas
+'''
+from spyder.app import start
+
+
+def hello(name,country='Finland'):
+    print("Hello {name}, how are things in {country}".format(name=name,country=country))
```
Comment (wmvanvliet) [inline gizmo.py]: alternatively:

```python
print(f"Hello {name}, how are things in {country}?")
```

File: gizmo.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    
+
+def spell(word):
+    spelled = ''
+    for i in range(len(word)-1):
+        spelled += word[i]+'.'
```
Comment (wmvanvliet) [inline gizmo.py]: alternatively:

```python
for letter in word[:-1]:
    spelled += letter + '.'
```

File: gizmo.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    print(spelled + word[-1])
+
+def relative_path(subjects, basepath='./subjects/mock_recording_', extension='.rec' ):
+    return [basepath + subj + extension for subj in subjects]
```
Comment (wmvanvliet) [inline gizmo.py]: you were asked to use python's formatting syntax here:

```python
return [f'./subjects/mock_recording_{subj}.rec' for subj in subjects]
```

Also, designing `basepath` to be a combination of folder and part of a filename is a weird design choice.

File: gizmo.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        
+        generate_fibonacci_sequence(maxcount)
+        
+        The returned generator yields fibonacci numbers up to the maximum number indicated.
+        Note, that at least 2 numbers are always yielded!
+        
+        Parameters
+        ----------
+        maxcount: int
+                  The number of fibonacci numbers to be yielded by the generator.
```
Comment (wmvanvliet) [inline gizmo.py]: We usually just indent with 4 spaces here:

```
maxcount : int
    The number of Fibonacci numbers to be yielded by the generator.
```

File: gizmo.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        
+        Yields
+        -------        
+        fibonacci: int
+                   The n-th number yielded is the n-th fibonacci number  
+         
+        """
+        last = 0
+        yield last
+        start = 1
```
Comment (wmvanvliet) [inline gizmo.py]: why `start` as a variable name? This is not the start of the sequence...

**Review by wmvanvliet — CHANGES_REQUESTED** [review]
Great start, almost there! Sorry about the workflow not running automatically for you. Hopefully I fixed that now.

File: gizmo.py (hunk 4 above)
Comment (tpfau) [reply gizmo.py]: Admittedly didn't know and the inital (variable names x,y) examples had the description indented at the same position as the type.

File: gizmo.py (hunk 3 above)
Comment (tpfau) [reply gizmo.py]: Should be updated

**Comment by wmvanvliet** [comment]
Sweet! Well done.

**Pull request closed without merging**